src/coreclr/src/pal/src/exception/seh-unwind.cpp

```diff
@@ -215,6 +254,13 @@ void UnwindContextToWinContext(unw_cursor_t *cursor, CONTEXT *winContext)
     unw_get_reg(cursor, UNW_AARCH64_X26, (unw_word_t *) &winContext->X26);
     unw_get_reg(cursor, UNW_AARCH64_X27, (unw_word_t *) &winContext->X27);
     unw_get_reg(cursor, UNW_AARCH64_X28, (unw_word_t *) &winContext->X28);
+
+#if defined(TARGET_OSX) && defined(TARGET_ARM64)
+    // Strip pointer authentication bits which seem to be leaking out of libunwind
+    // Seems like ptrauth_strip() / __builtin_ptrauth_strip() should work, but currently
```

Comment:
My suspicion is the kernel is doing the signing and the signing feature is only enabled for kernel code.

I am not sure why the bits aren't stripped regardless, but I wonder if it is because we are unwinding a foreign thread for hijacking it.  So our thread's SP is used to validate the pauth, but fails so the bits are left intact...

Clearly something we need to ask Apple about.

Reply:
I think it was a pretty common issue in tests which were hijacking threads to do garbage collection.

I opened #42955 to track fixing this.